src/js/components/input/AutomagicFormiflyField.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import {useFormiflyContext} from '../meta/FormiflyContext';
+import FormiflyCheckField from './FormiflyCheckField';
+import FormiflyField from './FormiflyField';
+import FormiflyMultiSelectField from './FormiflyMultiSelectField';
+import FormiflyRadioGroup from './FormiflyRadioGroup';
+import FormiflySelectField from './FormiflySelectField';
+import {StyledComponent} from 'styled-components';
+
+const AutomagicFormiflyField = (props: AutomagicFormiflyFieldProps) => {
+    const {getFieldProps} = useFormiflyContext();
+    const fieldProps = getFieldProps(props.name, props.help, props.type, props.value, props.id, props.additionalDescribedBy);
+    const type = props.type ?? fieldProps.type;
+
+    if (!!props.options) {
+        if (props.type === 'radio-group') {
+            // @ts-expect-error for some reason typescript doesn't understand that we checked for the existence of 'options'
+            return <FormiflyRadioGroup {...fieldProps} {...props}/>;
+        } else if (!!props.multiple || !!fieldProps.multiple) {
+            return <FormiflyMultiSelectField {...fieldProps} {...props}/>;
+        } else {
+            return <FormiflySelectField {...fieldProps} {...props}/>;
+        }
+    } else if (['checkbox', 'radio'].includes(type) && (fieldProps.checked !== undefined || props.checked !== undefined)) {
+        // @ts-expect-error for some reason typescript doesn't understand that we checked for the existence of 'checked'
+        return <FormiflyCheckField {...fieldProps} {...props}/>;
+    } else {
+        return <FormiflyField {...fieldProps} {...props}/>;
+    }
+};
+
+export type AutomagicFormiflyFieldProps = {
+    label: string,
+    name: string,
+    help?: string,
+    type?: string,
+    id?: string,
+    className?: string,
+    inputClassName?: string,
+    labelClassName?: string,
+    errorClassName?: string,
+    helpClassName?: string,
+    legendClassName?: string,
+    containerComponent?: StyledComponent<any, any>,
```

Comment:
Would this cause an error if someone used a non styled component for it in their consuming code?

Reply:
good question. Probably? I can try to find a better type. Gotta be honest, I'm wasn't entirely sure about many react types and had to guess a lot. But what I know believe to understand, I would assume that `React.ComponentType` would be a better type?